enterprise/frontend/src/embedding-sdk/bin/generate-nextjs-compat.ts

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable no-console */
+import fs from "fs";
+
+import path from "path";
+import prettier from "prettier";
+import { match } from "ts-pattern";
+
+type ComponentDefinition = {
+  mainComponent: string;
+  subComponents: string[];
+};
+
+// START OF CONFIGURATION
+// Note: this list needs to be updated when new components are added
+
+const COMPONENTS_TO_EXPORT: ComponentDefinition[] = [
+  // eslint-disable-next-line no-literal-metabase-strings -- cli tool
+
+  // MetabaseProvider is added manually because it needs to render children while loading
+  // we may have other components that need to render children while loading, in that case we can add a flag here
+  // { mainComponent: "MetabaseProvider", subComponents: [] },
+  { mainComponent: "StaticQuestion", subComponents: [] },
+  {
+    mainComponent: "InteractiveQuestion",
+    subComponents: [
+      "BackButton",
+      "FilterBar",
+      "Filter",
+      "FilterButton",
+      "FilterPicker",
+      "ResetButton",
+      "Title",
+      "Summarize",
+      "SummarizeButton",
+      "Editor",
+      "Notebook",
+      "Editor",
+      "NotebookButton",
+      "EditorButton",
+      "QuestionVisualization",
+      "SaveQuestionForm",
+      "SaveButton",
+      "ChartTypeSelector",
+      "EditorViewControl",
+      "QuestionSettings",
+    ],
+  },
+  {
+    mainComponent: "StaticDashboard",
+    subComponents: [],
+  },
+  { mainComponent: "InteractiveDashboard", subComponents: [] },
+];
+
+// END OF CONFIGURATION
+
+// eslint-disable-next-line no-literal-metabase-strings -- it's code
+const MetabaseProviderCode = `
+const MetabaseProvider = ({
+  config,
+  children,
+}) => {
+  const Provider = dynamic(
+    () =>
+      import("@metabase/embedding-sdk-react").then((m) => {
+        return { default: m.MetabaseProvider };
+      }),
+    {
+      ssr: false,
+      loading: () => {
+        return React.createElement("div", { id: "metabase-sdk-root" }, children);
```

Comment:
this is to make it render the children while loading the sdk, so that the application is still rendered while the app loads the sdk